internal/elasticsearch/client.go

```diff
@@ -108,5 +120,114 @@ func Client(customOptions ...ClientOption) (*elasticsearch.Client, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "can't create instance")
 	}
-	return client, nil
+	return &Client{Client: client}, nil
+}
+
+// CheckHealth checks the health of the cluster.
+func (client *Client) CheckHealth(ctx context.Context) error {
+	resp, err := client.Cluster.Health(client.Cluster.Health.WithContext(ctx))
+	if err != nil {
+		return errors.Wrap(err, "error checking cluster health")
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "error reading cluster health response")
+	}
+
+	var clusterHealth struct {
+		Status string `json:"status"`
+	}
+	err = json.Unmarshal(body, &clusterHealth)
+	if err != nil {
+		return errors.Wrap(err, "error decoding cluster health response")
+	}
+
+	if status := clusterHealth.Status; status != "green" && status != "yellow" {
+		if status != "red" {
+			return errors.Errorf("cluster in unhealthy state: %q", status)
+		}
+		cause, err := client.redHealthCause(ctx)
+		if err != nil {
+			return errors.Wrapf(err, "cluster in unhealthy state, failed to identify cause")
+		}
+		return errors.Errorf("cluster in unhealthy state: %s", cause)
```

Comment:
I was thinking if it would be helpful to diagnose further to include here something like:
> More info running inside the elasticsearch container: curl -k "https://localhost:9200/_internal/_health?pretty"

No need to run this command inside the elasticsearch contianer.

Reply:
Umm, this `cause` is obtained from `_internal/_health`. From all the information there I am only collecting the causes of impacts with highest severity. Should we collect something else?

Reply:
Just re-thinking about it, probably there is no need to add that message.

Let's consider that there are two "red" issues in the Elasticsearch cluster with different severity levels. In the first run, the cause of the issue with the highest severity would be shown to the user. Once solved, in the second run/attempt would be shown the other issue with the lowest severity if it is still active.

Taking into account that, I would say this could be a good way of handling these errors/issues. Displaying the issue with the highest severity. :+1: So, just dismiss this.

Moreover, in the case of using `elastic-package` in CI that message would not be helpful at all.

Reply:
Well, I could ignore the severity and just return all the causes. But for the example of the disk, it would also append the cause for shard unavailability, that would be redundant. We may consider a different approach if we find other cases to handle.